zone/mob_movement_manager.cpp

```diff
@@ -810,11 +810,55 @@ void MobMovementManager::UpdatePathGround(Mob * who, float x, float y, float z,
 	}
 
 	AdjustRoute(route, who);
+	   
+	
+	
+	//avoid doing any processing if the mob is stuck to allow normal stuck code to work.
+	if (!stuck)
+	{
+
+		//there are times when the routes returned are no differen than where the mob is currently standing. What basically happens
+		//is a mob will get 'stuck' in such a way that it should be moving but the 'moving' place is the exact same spot it is at. 
+		//this is a problem and creates an area of ground that if a mob gets to, will stay there forever. If socal this creates a 
+		//"Ball of Death" (tm). This code tries to prevent this by simply warping the mob to the requested x/y. Better to have a warp than 
+		//have stuck mobs.
+
+		auto routeNode = route.begin();
+		bool noValidPath = true;
+		while (routeNode != route.end() && noValidPath == true) {
+			auto &currentNode = (*routeNode);
+
+			if (routeNode == route.end())
+			{
+				continue;
+			}
+
+			if (!(currentNode.pos.x == who->GetX() && currentNode.pos.y == who->GetY()))
```

Comment:
Should use IsPositionEqual or IsPositionEqualWithinCertainZ instead of directly comparing the float values.  This handles slight variations from the float values counting against equality.  (It happens a lot more than you'd think).

Reply:
While I agree on a technical level about floats and their direct comparisons, the reason I didn't is the X/Y were set by this function to begin with thus it will be exactly the same as they try and give the exact positions again.